MIGRATION.md

```diff
@@ -0,0 +1,413 @@
+```
+
+The server host and port can be specified in `server()` if needed, but by default, the server binds to `localhost` on a random available port. More details can be found in the [API reference][pact.pact.Pact.serve].
+
+##### Writing Pact Files
+
+Since the old v2 API executed a sub-process for the mock service, the Pact file was automatically written when the context manager exited or when `pact.verify()` was called. The new v3 API runs the mock service in-process, so the Pact file must be written explicitly using the `write_file()` method:
+
+```python title="v2"
+with pact:
+    # tests...
+# Pact file written automatically
+```
+
+```python title="v3"
+pact = Pact('consumer', 'provider')
+# Define interactions and run tests...
+pact.write_file('/path/to/pacts')
+```
+
+#### Matchers
+
+Support for matchers has been greatly expanded and improved in the v3 API. The older v2 classes defined a limited set of matchers, while the new API provides a more comprehensive and flexible approach.
+
+```python title="v2"
+from pact.v2.matchers import Like, EachLike, Regex, Term
+
+# Usage:
+Like({'id': 123})
+EachLike({'item': 'value'})
+Regex('hello world', r'^hello')
+```
+
+```python title="v3"
+from pact import match
+
+# Usage:
+match.like({'id': 123})
+match.each_like({'item': 'value'})
+match.regex('hello world', r'^hello')
+```
+
+For a full list of available matchers and their usage, refer to the [API documentation][pact.match].
+
+### Provider Changes
+
+The provider verification API has been completely redesigned in v3 to provide a more intuitive and flexible interface. The old `Provider` and `Verifier` classes have been replaced by a single `Verifier` class with a fluent interface.
+
+#### Creating a Verifier
+
+```python title="v2"
+from pact.v2 import Provider, Verifier
+
+# Create separate Provider and Verifier instances
+provider = Provider('my-provider')
+verifier = Verifier(provider, 'http://localhost:8080')
+```
+
+```python title="v3"
+from pact import Verifier
+
+# Single Verifier instance with provider name
+verifier = Verifier('my-provider')
+```
+
+The protocol specification is now done through the `add_transport` method, which allows for more flexible configuration and supports multiple transports if needed.
+
+```python title="v2"
+verifier = Verifier(provider, 'http://localhost:8080')
+```
+
+```python title="v3"
+verifier = (
+    Verifier('my-provider')
+    .add_transport(url='http://localhost:8080')
+    # Or more granular control:
+    .add_transport(
+        protocol='http',
+        port=8080,
+        path='/api/v1',
+        scheme='https'
+    )
+)
+```
+
+#### Adding Pact Sources
+
+Support for both local files and Pact Brokers is retained in v3, with the `verify_pacts` and `verify_with_broker` methods replaced by a more flexible source configuration. This allows multiple sources to be combined, and selectors to be applied.
+
+<!-- markdownlint-disable code-block-style -->
+
+=== "Local Files"
```

Comment:
can this be a url , for verifications by webhook?

how do we add auth in that case?

Reply:
I have no tested the webhook flow at all... so I'll take a look and see.